### ISSUE TITLE
"Use of uninitialized value in numeric le" warning when executing Build file

### ISSUE THREAD

**Issue by skaji** [Leont/dist-build#1]
I got `Use of uninitialized value in numeric le` warning
when executing Build powered by Dist::Build.

### my environment

* macos 14.5
* perl 5.40.0
* Dist::Build 0.003
* ExtUtils::Builder::Plan 0.006

### steps to reproduce

reproduce.sh:
```bash
#!/bin/bash

set -eux

rm -rf Foo

mkdir Foo
cd Foo
mkdir lib

cat << 'EOF' > lib/Foo.pm
package Foo;
our $VERSION = '0.01';
1;
EOF

perl -MCPAN::Meta -e 'CPAN::Meta->new({ version => "0.01", name => "Foo" })->save("META.json")'

cat << 'EOF' > Build.PL
use strict;
use warnings;
use Dist::Build;
Build_PL(\@ARGV, \%ENV);
EOF

perl Build.PL
./Build
```

```console
❯ bash reproduce.sh
+ rm -rf Foo
+ mkdir Foo
+ cd Foo
+ mkdir lib
+ cat
+ perl -MCPAN::Meta -e 'CPAN::Meta->new({ version => "0.01", name => "Foo" })->save("META.json")'
+ cat
+ perl Build.PL
Creating new 'Build' script for 'Foo' version '0.01'
+ ./Build
cp lib/Foo.pm blib/lib/Foo.pm
Use of uninitialized value in numeric le (<=) at /Users/skaji/env/plenv/versions/5.40.0/lib/perl5/site_perl/5.40.0/ExtUtils/Builder/Plan.pm line 52.
```

**Comment by Leont** [comment]
I can't reproduce this issue on Linux. I'm not really having enough information right now to diagnose anything right now, other than saying the issue is probably in `ExtUtils::Builder::Plan`

**Comment by skaji** [comment]
> I can't reproduce this issue on Linux.

You are right. This issue seems to occur only on macOS.

**Comment by skaji** [comment]
OK, macos filesystem is case-insensitive,
so `-e $name` for $name = "build" is true (because there is `Build` file).
https://metacpan.org/dist/ExtUtils-Builder/source/lib/ExtUtils/Builder/Plan.pm#L52

**Comment by Leont** [comment]
> OK, macos filesystem is case-insensitive,
so -e $name for $name = "build" is true (because there is Build file).
https://metacpan.org/dist/ExtUtils-Builder/source/lib/ExtUtils/Builder/Plan.pm#L52

That almost explains what's going on, but the `build` action should be phony, so that `-e` should never trigger anyway.

**Comment by Leont** [comment]
I found the issue, there was a typo in the serializer that caused all nodes to be non-phony. Will release a fix shortly.

**Comment by Leont** [comment]
Fixed in Dist::Build 0.004
